scipy/stats/stats.py

```diff
@@ -3565,6 +3568,159 @@ def count_rank_tie(ranks):
     # Limit range to fix computational errors
     return KendalltauResult(min(1., max(-1., tau)), pvalue)
 
+WeightedTauResult = namedtuple('WeightedTauResult', ('correlation', 'pvalue'))
+
+
+def weightedtau(x, y, rank=True, weigher=lambda x: 1./(x+1), additive=True):
```

Comment:
Good point. I added a

    if weigher == False:
        weigher = lambda x: 1./(1 + x)

at the start of the Cython method, and it's more than twice faster now for the hyperbolic case.